app/src/main/java/com/rootstrap/android/util/permissions/PermissionActivity.kt

```diff
@@ -0,0 +1,70 @@
+package com.rootstrap.android.util.permissions
+
+import android.content.pm.PackageManager
+import androidx.core.app.ActivityCompat
+import com.rootstrap.android.ui.base.BaseActivity
+
+open class PermissionActivity : BaseActivity() {
+
+    private var permissionListener: PermissionResponse? = null
+
+    private fun requestPermission(permissions: Array<String>) {
+        val notGrantedPermissions = this.checkNoGrantedPermissions(permissions)
+
+        when {
+            notGrantedPermissions.isEmpty() -> permissionListener?.granted()
+            else -> ActivityCompat.requestPermissions(
+                this,
+                notGrantedPermissions.toTypedArray(),
+                REQUEST_PERMISSION_REQUEST_CODE
+            )
+        }
+    }
+
+    fun checkPermission(permission: String, listener: PermissionResponse) {
+        permissionListener = listener
+        requestPermission(arrayOf(permission))
```

Comment:
I'd call `checkPermissions(arrayOf(permission), listener)` to keep the logic in one place.